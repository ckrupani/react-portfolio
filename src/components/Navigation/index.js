import React from "react";

function Navigation({ menuItems, activeItem, setActiveItem }) {
    return (
        <>
            <h2 className="header">
                <a data-testid="link" href="/">
                    Chirag
                </a>
            </h2>
            <nav>
                <ul className="flex-row">
                    {menuItems.map((item, i) => (
                        <li
                            key={item.name}
                            className={`mx-2 ${
                                activeItem.name === item.name && "navActive"
                            }`}
                            onClick={() => setActiveItem(item)}
                        >
                            <span>{item.title}</span>
                        </li>
                    ))}
                </ul>
            </nav>
        </>
    );
}

export default Navigation;
