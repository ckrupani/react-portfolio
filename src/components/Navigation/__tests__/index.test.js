import React from "react";
import { render, cleanup, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Navigation from "..";

afterEach(cleanup);

describe("Navigation component", () => {
    const menuItems = [
        {
            name: "about",
            title: "About me",
        },
        { name: "portfolio", title: "Portfolio" },
        { name: "contact", title: "Contact" },
        {
            name: "resume",
            title: "Resume",
        },
    ];
    const activeItem = {
        name: "about",
        title: "About me",
    };
    const setActiveItem = jest.fn();

    it("renders", () => {
        render(
            <Navigation
                menuItems={menuItems}
                activeItem={activeItem}
                setActiveItem={setActiveItem}
            />
        );
    });

    it("matches snapshot DOM node structure", () => {
        const { asFragment } = render(
            <Navigation
                menuItems={menuItems}
                activeItem={activeItem}
                setActiveItem={setActiveItem}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("inserts the name of the developer", () => {
        render(
            <Navigation
                menuItems={menuItems}
                activeItem={activeItem}
                setActiveItem={setActiveItem}
            />
        );

        expect(screen.getByTestId("link")).toHaveTextContent("Chirag");
    });
});
