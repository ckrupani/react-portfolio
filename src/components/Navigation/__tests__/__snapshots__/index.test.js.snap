// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <h2
    class="header"
  >
    <a
      data-testid="link"
      href="/"
    >
      Chirag
    </a>
  </h2>
  <nav>
    <ul
      class="flex-row"
    >
      <li
        class="mx-2 navActive"
      >
        <span>
          About me
        </span>
      </li>
      <li
        class="mx-2 false"
      >
        <span>
          Portfolio
        </span>
      </li>
      <li
        class="mx-2 false"
      >
        <span>
          Contact
        </span>
      </li>
      <li
        class="mx-2 false"
      >
        <span>
          Resume
        </span>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
